### ISSUE TITLE
when calling https on locahost I get "Cannot read property 'status' of undefined"

### ISSUE THREAD

**Issue by royts** [visionmedia/supertest#370]
Hi,

here is my test code:

```
var request = require('supertest');
describe("test", function () {
  it("should pass", function (done) {
    request("https://localhost:8443")
      .post("/path")
      .send({})
      .expect(200)
      .expect({status: 'success'}, done);
  });
});
```

when I run it get the error:

```
TypeError: Cannot read property 'status' of undefined
      at Test._assertStatus (node_modules/supertest/lib/test.js:247:10)
      at Test._assertFunction (node_modules/supertest/lib/test.js:265:11)
      at Test.assert (node_modules/supertest/lib/test.js:153:18)
      at assert (node_modules/supertest/lib/test.js:131:12)
      at node_modules/supertest/lib/test.js:128:5
      at Test.Request.callback (node_modules/superagent/lib/node/index.js:587:12)
      at ClientRequest.<anonymous> (node_modules/superagent/lib/node/index.js:539:10)
      at TLSSocket.socketErrorListener (_http_client.js:306:9)
      at emitErrorNT (net.js:1272:8)
      at _combinedTickCallback (internal/process/next_tick.js:74:11)
      at process._tickCallback (internal/process/next_tick.js:98:9)
```

while with curl I get a valid answer:

```
curl -d "" https://localhost:8443/path -k
```

the answer:

```
{"status":"success"}%
```

_any idea what am I doing wrong?_

btw - I'm using latest version (2.0.0)

thanks.

**Comment by equicolor** [comment]
+1

**Comment by piemme** [comment]
+1

**Comment by royts** [comment]
Putting `process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';` before the test solved it for me.

I'll create a pull request to add it to the docs.

**Comment by mikelax** [comment]
@royts do you want to upgrade to 2.0.1 and see if that resolves the issue?

**Comment by cagdastulek** [comment]
+1

Having same issue after upgrading from 1.0.1 to 2.0.1 without any code changes. 

Neither NODE_TLS_REJECT_UNAUTHORIZED nor 2.0.1 helps.

**Comment by jlindsay** [comment]
+1
process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0' does does help.  

at **Test.< anonymous >** (src/xxxxx.js:81:29)
      at Test.assert (node_modules/supertest/lib/test.js:156:6)
      at assert (node_modules/supertest/lib/test.js:127:12)
      at node_modules/supertest/lib/test.js:124:5
      at Test.Request.callback (node_modules/superagent/lib/node/index.js:687:12)
      at ClientRequest.<anonymous> (node_modules/superagent/lib/node/index.js:639:10)
      at TLSSocket.socketErrorListener (_http_client.js:265:9)
      at connectErrorNT (net.js:996:8)

**Comment by nathanhornby** [comment]
> process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';

Fixed it for me, thanks! Maybe SSL could be a boolean option passed in somewhere?

**Comment by bkeepers** [comment]
https://github.com/visionmedia/supertest/pull/446 should help reveal the underlying error here.

**Comment by mikelax** [comment]
V3.1.0 has been published which includes #446 to solve handling for this error case.
